plugins/PostProcessingPlugin/scripts/DisplayInfoOnLCD.py

```diff
@@ -1,42 +1,62 @@
+ Modified: Joshua Pope-Lewis on November 16, 2018
+ Display Progress on LCD by Mathias Lyngklip Kjeldgaard, Alexander Gee, Kimmo Toivanen, Inigo Martinez on July 31, 2019
+ Show Progress was adapted from Display Progress by Louis Wooters on January 6, 2020.  His changes are included here.
+---------------------------------------------------------------
+ DisplayNameOrProgressOnLCD.py
+ Cura Post-Process plugin
+ Combines 'Display Filename and Layer on the LCD' with 'Display Progress'
+ Combined and with additions by: GregValiant (Greg Foresi)
+ Date:  September 8, 2023
+ Date:  March 31, 2024 - Bug fix for problem with adding M118 lines if 'Remaining Time' was not checked.
+ NOTE:  This combined post processor will make 'Display Filename and Layer on the LCD' and 'Display Progress' obsolete
+ Description:  Display Filename and Layer options:
+       Status messages sent to the printer...
+           - Scrolling (SCROLL_LONG_FILENAMES) if enabled in Marlin and you aren't printing a small item select this option.
+           - Name: By default it will use the name generated by Cura (EG: TT_Test_Cube) - You may enter a custom name here
+           - Start Num: Choose which number you prefer for the initial layer, 0 or 1
+           - Max Layer: Enabling this will show how many layers are in the entire print (EG: Layer 1 of 265!)
+           - Add prefix 'Printing': Enabling this will add the prefix 'Printing'
+           - Example Line on LCD:  Printing Layer 0 of 395 3DBenchy
+       Display Progress options:
+           - Display Total Layer Count
+           - Disply Time Remaining for the print
+           - Time Fudge Factor % - Divide the Actual Print Time by the Cura Estimate.  Enter as a percentage and the displayed time will be adjusted.
+                This allows you to bring the displayed time closer to reality (Ex: Entering 87.5 would indicate an adjustment to 87.5% of the Cura estimate).
+           - Example line on LCD:  1/479 | ET 2h13m
+           - Time to Pauses changes the M117/M118 lines to countdown to the next pause as  1/479 | TP 2h36m
+           - 'Add M118 Line' is available with either option.  M118 will bounce the message back to a remote print server through the USB connection.
+           - 'Add M73 Line' is used by 'Display Progress' only.  There are options to incluse M73 P(percent) and M73 R(time remaining)
+           - Enable 'Finish-Time' Message - when enabled, takes the Print Time and calculates when the print will end.  It uses the Time Fudge Factor.  The user may enter a print start time.
+Date:  June 30, 2025 Cost of electricity added to the other print statistics in '_add_stats'.
+"""
 
 from ..Script import Script
 from UM.Application import Application
 from UM.Qt.Duration import DurationFormat
 import time
 import datetime
 import math
+from UM.Preferences import Preferences
 from UM.Message import Message
+from UM.Logger import Logger
```

Comment:
```suggestion
from UM.Message import Message
```
Imports are not used currently so they can be removed.

Reply:
Done.